service_configuration_lib/spark_config.py

```diff
@@ -449,9 +451,46 @@ def _cap_executor_resources(user_spark_opts: Dict[str, str]) -> Dict[str, str]:
         )
         executor_cores = EXECUTOR_CORES_CAP
 
+    return executor_cores, executor_memory
+
+
+def _recalculate_executor_resources(
+    user_spark_opts: Dict[str, str],
+    force_spark_resource_configs: bool,
+) -> Dict[str, str]:
+    executor_cores = int(user_spark_opts.get('spark.executor.cores', str(DEFAULT_EXECUTOR_CORES)))
+    executor_memory = user_spark_opts.get('spark.executor.memory', DEFAULT_EXECUTOR_MEMORY)
+    executor_instances = int(user_spark_opts.get('spark.executor.instances', str(DEFAULT_EXECUTOR_INSTANCES)))
+
+    memory_mb = parse_memory_string(executor_memory)
+
+    if force_spark_resource_configs:
```

Comment:
Let's add a warning here that it's generally not recommended to use `--force-spark-resource-configs` as it can result in non-optimal bin-packing of executors on aws nodes or can lead to wastage the resources. Please make sure there is a reason that this flag needs to be used.